File: cpp/121_BestTimetoBuyandSellStock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int len = prices.size();
        int maxPro = 0;
        int low = INT_MAX;
        for(int i = 0; i < len; ++i){
            low = std::min(prices[i] ,low);
            maxPro = std::max(prices[i] - low , maxPro);
        }
        return maxPro;
    }
};

File: js/121_BestTimetoBuyandSellStock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    var maxPro = 0;
    var low = Number.MAX_SAFE_INTEGER;
    for(var i = 0; i < prices.length; ++i){
        low = Math.min(prices[i], low);
        maxPro = Math.max(prices[i] - low, maxPro);
    }
    return maxPro;
};